core/store/src/main/java/uk/gov/gchq/gaffer/store/operation/handler/function/FunctionValidator.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2017 Crown Copyright
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package uk.gov.gchq.gaffer.store.operation.handler.function;
+
+import uk.gov.gchq.gaffer.commonutil.stream.Streams;
+import uk.gov.gchq.gaffer.data.element.function.ElementAggregator;
+import uk.gov.gchq.gaffer.data.element.function.ElementFilter;
+import uk.gov.gchq.gaffer.data.element.function.ElementTransformer;
+import uk.gov.gchq.gaffer.operation.impl.function.Aggregate;
+import uk.gov.gchq.gaffer.operation.impl.function.Filter;
+import uk.gov.gchq.gaffer.operation.impl.function.Function;
+import uk.gov.gchq.gaffer.operation.impl.function.Transform;
+import uk.gov.gchq.gaffer.operation.util.AggregatePair;
+import uk.gov.gchq.gaffer.store.schema.Schema;
+import uk.gov.gchq.gaffer.store.schema.SchemaEdgeDefinition;
+import uk.gov.gchq.gaffer.store.schema.SchemaElementDefinition;
+import uk.gov.gchq.gaffer.store.schema.SchemaEntityDefinition;
+import uk.gov.gchq.koryphe.ValidationResult;
+import uk.gov.gchq.koryphe.binaryoperator.AdaptedBinaryOperator;
+import uk.gov.gchq.koryphe.signature.Signature;
+import uk.gov.gchq.koryphe.tuple.binaryoperator.TupleAdaptedBinaryOperator;
+import uk.gov.gchq.koryphe.tuple.function.TupleAdaptedFunction;
+import uk.gov.gchq.koryphe.tuple.predicate.TupleAdaptedPredicate;
+
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.BinaryOperator;
+import java.util.stream.Collectors;
+
+public class FunctionValidator<T extends Function> {
+    public ValidationResult validate(final T operation, final Schema schema) {
+        final ValidationResult result = new ValidationResult();
+        if (null == operation) {
+            result.addError("Operation cannot be null.");
+        }
+        if (null == schema) {
+            result.addError("Schema cannot be null.");
+        }
+
+        result.add(validateOperation(operation, schema));
+
+        return result;
+    }
+
+    private ValidationResult validateOperation(final T operation, final Schema schema) {
+        final ValidationResult result = new ValidationResult();
+        final boolean opIsAggregate = operation instanceof Aggregate;
```

Comment:
Rather than doing this, it would be better to have separate classes: AggregateValidator, FilterValidator and TransformFilter, which extends FunctionValidator.